### DISCUSSION TITLE
milvus-querynode replicas

### DISCUSSION THREAD

**Discussion by chensanle** [milvus-io/milvus#15567] [Q&A and General discussion]
The milvus-querynode Deployment set with 10 replicas, but only 2 replicas are working, how do I use the remaining 8 replicas?
![image](https://user-images.githubusercontent.com/31087327/153814397-ee5b29a9-56a3-4e1b-b75a-483352c474d9.png)

**Comment by yhmo** [comment]
How many collections are in the cluster?
How many rows are in each collection?
Milvus will balance segments between query nodes, if the segments are too few, some of the query nodes will be idle.

**Reply by chensanle** [reply]
Thanks for your apply.
collections: 1 (ann_1m_sq8)
rows: 1M 
machine: 32c64g * 2

I have 100 threads to request the milvus, each request cost about 500 ms, but only two or less nodes at work. I know if all the querynodes work, the time cost may decrease. How can I optimize the program？

**Reply by yhmo** [reply]
By default, each collection has only two sharding channels, which means inserted data will be split into two parts. If the data size is less than segment size, so the 1M rows were split into two segments. This is why only two query nodes are busy. In the create collection methods you can find the parameter shard_num=2. You can create a new collection with shard_num=[query nodes count] and try again.

And for search performance, please make sure to specify a "guarantee_timestamp" not equal 0, other the milvus server will have 200ms delay:
```
search_param = {
    "data": search_vectors,
    "anns_field": vector_field,
    "param": {"metric_type": METRIC_TYPE, "params": {"nprobe": NPROBE}},
    "limit": TOPK,
    "expr": "id_field > 0"
    "guarantee_timestamp": 1}
results = collection.search(**search_param)
```

**Reply by chensanle** [reply]
Thanks for your clear and elegant reply.
Firstly, the guarantee_timestamp already has been set to 1.
It seems how many sharding I have and how many workers will work, Bug if I have to access some data which not so large at a extremely high frequency, there also can be only two workers(two sharding)?

**Reply by xiaofan-luan** [reply]
> Thanks for your clear and elegant reply. Firstly, the guarantee_timestamp already has been set to 1. It seems how many sharding I have and how many workers will work, Bug if I have to access some data which not so large at a extremely high frequency, there also can be only two workers(two sharding)?

Not really. sharding only affects ingestion rate. If data converted from growing segment to sealed segment, it will be balanced to other querynodes.

The problem will be if your data is extremely small and your qps is high. Then you will need the memory replica to spread your data into multi copys, the memory replica will be released at milvus 2.1~

**Reply by chensanle** [reply]
querynode cpu resource config:
resource required: 0.5; resource limit: 10

When I continuoue to query the result with 100 threads from milvus. However the [cpu load ](https://github.com/milvus-io/milvus/discussions/15567#discussion-3875193) is lower than expected and many requests seems waiting in line. Do milvus has any cpu restrictions for one querynode, or there are restrictions on concurrent requests?   

By the way，how can I find the milvus system status tab in attu?

**Reply by yhmo** [reply]
Did you try click the "System view" on the Attu page?

**Reply by chensanle** [reply]
@yhmo Where can i find the "System view" button?
![image](https://user-images.githubusercontent.com/31087327/154252953-e3170214-e6da-45f1-b000-12c3442725b3.png)

**Reply by yhmo** [reply]
There should be a button here:
![11111](https://user-images.githubusercontent.com/2282099/154390933-0a23c6a4-3e56-408a-9c87-9ac4db691304.png)

You can try the latest attu:
`docker pull zilliz/attu:v2.0.1`

**Reply by xiaofan-luan** [reply]
@shanghaikid please help myman

**Reply by shanghaikid** [reply]
@xiaofan-luan please update to the latest version.
